utility/src/main/java/org/oppia/android/util/logging/performancemetrics/PerformanceMetricsAssessorImpl.kt

```diff
@@ -70,4 +77,48 @@ class PerformanceMetricsAssessorImpl @Inject constructor(
       else -> HIGH_MEMORY_TIER
     }
   }
+
+  override fun computeCpuSnapshotAtCurrentTime(): CpuSnapshot {
+    return CpuSnapshot(
+      appTimeMillis = oppiaClock.getCurrentTimeMs(),
+      cpuTimeMillis = Process.getElapsedCpuTime(),
+      numberOfOnlineCores = getNumberOfOnlineCores()
+    )
+  }
+
+  override fun getRelativeCpuUsage(
+    firstCpuSnapshot: CpuSnapshot,
+    secondCpuSnapshot: CpuSnapshot
+  ): Double? {
+    if (
+      firstCpuSnapshot.isNewer(secondCpuSnapshot) ||
+      firstCpuSnapshot.doesNotHaveValidNumberOfOnlineCores() ||
+      secondCpuSnapshot.doesNotHaveValidNumberOfOnlineCores()
+    ) { return null }
+
+    val deltaCpuTimeMs = secondCpuSnapshot.cpuTimeMillis - firstCpuSnapshot.cpuTimeMillis
+    val deltaProcessTimeMs = secondCpuSnapshot.appTimeMillis - firstCpuSnapshot.appTimeMillis
+    val numberOfCores =
+      secondCpuSnapshot.numberOfOnlineCores + firstCpuSnapshot.numberOfOnlineCores / 2.0
+    return when (val relativeCpuUsage = deltaCpuTimeMs / (deltaProcessTimeMs * numberOfCores)) {
+      in 0.0..1.0 -> relativeCpuUsage
+      else -> { null }
```

Comment:
```suggestion
      else -> null
```

(Braces can be omitted here).

Reply:
done.